R/utils.R

```diff
@@ -504,3 +504,175 @@ download_with_retry <- function(...) {
   }
   return(rc)
 }
+
+#' Checks if NONMEM run is done by looking for "Stop Time" in .lst file
+#'
+#' Returns `TRUE` if the model appears to be finished running and `FALSE` otherwise.
+#' @param mod either a bbi_nonmem_model object or a path to an .lst file
+#' @export
+check_nonmem_finished <- function(mod) {
```

Comment:
The more I think about it, I _don't_ think `check_nonmem_finished()` should be S3 dispatch that can optionally take the `.lst` path. I think it should only take the model object. You can just combine these into a single function and call [check_model_object(.mod, NM_MOD_CLASS)](https://github.com/metrumresearchgroup/bbr/blob/main/R/utils.R#L370-L379) at the top.

The reason is that almost all of our other functions insist on taking the model object, which is a decision we made early on to avoid path confusion ("do I pass the path to the YAML, to the .lst, I can't remember?...").

And to be clear, I'm just talking about `check_nonmem_finished()`. I think the S3 implementation of `wait_for_nonmem()` is a good idea.